modules/openapi-generator/src/main/resources/rust/reqwest/api.mustache

```diff
@@ -33,7 +33,7 @@ impl {{{classname}}} for {{{classname}}}Client {
         let configuration: &configuration::Configuration = self.configuration.borrow();
         let client = &configuration.client;
 
-        let uri_str = format!("{}{{{path}}}", configuration.base_path{{#pathParams}}, {{{baseName}}}={{#isString}}urlencode({{/isString}}{{{paramName}}}{{#isListContainer}}.join(",").as_ref(){{/isListContainer}}{{#isString}}){{/isString}}{{/pathParams}});
+        let uri_str = format!("{}{{{path}}}", configuration.base_path{{#pathParams}}, {{{baseName}}}={{#isString}}::apis::urlencode({{/isString}}{{{paramName}}}{{#isListContainer}}.join(",").as_ref(){{/isListContainer}}{{#isString}}){{/isString}}{{/pathParams}});
```

Comment:
Looking at https://github.com/OpenAPITools/openapi-generator/pull/3332, should this be `crate::apis...` instead for compatibility with Rust 2018?

Reply:
yeah, so `crate::apis::` will not work with Rust 2015. I started to switch import statements to the Rust 2018 format, but decided against it to avoid breaking changes. However, since it's already staged, I'll just wait for #3332 to be merged and rebase it.

Reply:
That sounds fine.

I thought `crate::apis::` _was_ compatible with Rust 2015, though? If I apply the following diff, the sample continues to compile successfully.

```diff
diff --git a/samples/client/petstore/rust-reqwest/Cargo.toml b/samples/client/petstore/rust-reqwest/Cargo.toml
index 6d66b8f7c7..25b15f54fd 100644
--- a/samples/client/petstore/rust-reqwest/Cargo.toml
+++ b/samples/client/petstore/rust-reqwest/Cargo.toml
@@ -2,6 +2,7 @@
 name = "petstore_client"
 version = "1.0.0"
 authors = ["OpenAPI Generator team and contributors"]
+edition = "2015"
 
 [dependencies]
 serde = "^1.0"
diff --git a/samples/client/petstore/rust-reqwest/src/apis/client.rs b/samples/client/petstore/rust-reqwest/src/apis/client.rs
index ea6b0eb635..5233b3cf5d 100644
--- a/samples/client/petstore/rust-reqwest/src/apis/client.rs
+++ b/samples/client/petstore/rust-reqwest/src/apis/client.rs
@@ -15,7 +15,7 @@ impl APIClient {
 
         APIClient {
             configuration: rc.clone(),
-            pet_api: Box::new(::apis::PetApiClient::new(rc.clone())),
+            pet_api: Box::new(crate::apis::PetApiClient::new(rc.clone())),
             store_api: Box::new(::apis::StoreApiClient::new(rc.clone())),
             user_api: Box::new(::apis::UserApiClient::new(rc.clone())),
         }
```

I've not done much around the transition between editions, so would not be surprised if I was misunderstanding what was going on here...

Reply:
Actually, you're totally right, I didn't realize this. I just fixed it in a commit on top, and regenerated the 2 clients. I think you can merge both this one and #3332 independently.